src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        copyState = { ...copyState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        copyState = { ...copyState };
+
+        for (const key of action.keysToRemove) {
+          delete copyState[key];
+        }
+        break;
+
+      case 'clear':
+        copyState = {};
```

Comment:
There is a missing `break` statement after the 'clear' case. Without it, the function might not behave as expected if more cases are added in the future. Consider adding `break;` after `copyState = {};`.